src/main/java/core/basesyntax/service/impl/FruitShopCalculationServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitShopStorage;
+import core.basesyntax.model.AbstractTransaction;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.ShopCalculationService;
+import java.util.List;
+
+public class FruitShopCalculationServiceImpl implements ShopCalculationService<Fruit> {
+    private final FruitShopStorage fruitShopStorage;
+
+    public FruitShopCalculationServiceImpl(FruitShopStorage fruitShopStorage) {
+        this.fruitShopStorage = fruitShopStorage;
+    }
+
+    @Override
+    public FruitShopStorage calculate(List<AbstractTransaction<Fruit>> abstractTransactions) {
```

Comment:
The `calculate` method currently returns a `FruitShopStorage`, which is specific to `Fruit`. To maintain consistency with the generic type `T` in the `ShopCalculationService` interface, consider changing the return type to `ShopStorage<Fruit>`. This will allow the implementation to be more flexible and align with the interface definition.